django/db/migrations/serializer.py

```diff
@@ -244,8 +244,9 @@ def _format(self):
 
 class SetSerializer(BaseSequenceSerializer):
     def _format(self):
-        # Don't use the literal "{%s}" as it doesn't support empty set
-        return "set([%s])"
+        # When value is empty, the empty set should be serialized as "set()",
+        # not "{}" because {} is an empty dict.
+        return '{%s}' if self.value else 'set(%s)'
```

Comment:
I think this actually makes the code look worse in this particular case. Is there any tangible benefit?